python/across_api/base/ephem.py

```diff
@@ -0,0 +1,193 @@
+        Sun/Moon position and latitude/longitude of the spacecraft
+        initially.
+
+        Returns
+        -------
+            True if successful, False if not.
+        """
+
+        # Check if all parameters are valid
+        if not self.validate_get():
+            # Compute Ephemeris
+            return False
+
+        # Check the TLE is available
+        if self.tle is None:
+            raise HTTPException(
+                status_code=404, detail="No TLE available for this epoch"
+            )
+
+        # Round start and end times to stepsize, create array of timestamps
+        begin = round_time(self.begin, self.stepsize)
+        end = round_time(self.end, self.stepsize)
+        steps = (end - begin).to(u.s) / (self.stepsize.to(u.s)) + 1
+        self.timestamp = Time(np.linspace(begin, end, steps))
+
+        # Load in the TLE data
+        satellite = Satrec.twoline2rv(self.tle.tle1, self.tle.tle2)
+
+        # Calculate TEME position and velocity for Satellite
+        _, temes_p, temes_v = satellite.sgp4_array(
+            self.timestamp.jd1, self.timestamp.jd2
+        )
+
+        # Convert SGP4 TEME data to astropy ITRS SkyCoord
+        teme_p = CartesianRepresentation(temes_p.T * u.km)
+        teme_v = CartesianDifferential(temes_v.T * u.km / u.s)
+        self.itrs = SkyCoord(
+            teme_p.with_differentials(teme_v), frame=TEME(obstime=self.timestamp)
+        ).itrs
+
+        # Set up astroplan Observer class
+        self.observer = Observer(self.itrs.earth_location)
+
+        # Calculate satellite position in GCRS coordinate system vector as
+        # array of x,y,z vectors in units of km, and velocity vector as array
+        # of x,y,z vectors in units of km/s
+        self.posvec, self.velvec = self.observer.location.get_gcrs_posvel(
+            self.timestamp
+        )
+
+        # Calculate the position of the Moon relative to the spacecraft
+        self.moon = get_body("moon", self.timestamp, location=self.observer.location)
+
+        # Calculate the position of the Moon relative to the spacecraft
+        self.sun = get_body("sun", self.timestamp, location=self.observer.location)
+
+        # Calculate the position of the Earth relative to the spacecraft
+        self.earth = get_body("earth", self.timestamp, location=self.observer.location)
+
+        # Calculate the latitude, longitude and distance from the center of the
+        # Earth of the satellite
+        self.longitude = self.observer.longitude
+        self.latitude = self.observer.latitude
+        dist = self.posvec.norm()
+
+        # Calculate the Earth radius in degrees
+        if self.earth_radius is not None:
+            self.earthsize = self.earth_radius * np.ones(len(self))
+        else:
+            self.earthsize = np.arcsin(R_earth / dist)
+
+        # Calculate orbit pole vector
+        polevec = self.posvec.cross(self.velvec)
+        self.pole = SkyCoord(polevec / polevec.norm())
```

Comment:
What do you need this for?

Reply:
For BurstCube, we don't. For other missions it's useful to know where the orbit poles are so you can calculate if a target is inside an orbit pole constraint (Swift has these) or for other missions if targets are inside a continuous viewing zone.

For BurstCube only, it can be removed.

Reply:
If we needed an orbit pole constraint, then I would suggest implementing that as a reusable constraint class.

Reply:
Never mind, this has to do with where code goes, which we agreed was off limits for review for now.

Reply:
Yeah, it's additional CPU time that's not needed for every mission, so makes sense to move it out of Ephem, especially for missions that don't need it. In the original code I only calculated it for missions that would use it, but in the spirit of code simplification I took that option out. I'll just remove it, no biggy.